[WIP] Some cleanups

mparkhe opened the pull request:
Getting some conformity with usage for RunStatus and source. Some paths were using strings while others were using the enum values.

- More testing with local MySQL backend
- Additional conformity between FileStore and SqlAlchemyStore (statuses and sources)
- Using constants from entities

Commit:
Search API: allowing param value to have any content for checks

Commit:
- Conformity between FileStore and SqlAlchemyStore
- Using constants from entities

Commit:
Merge branch 'master' of github.com:databricks/mlflow into search_fix

Commit:
enum to string fixes

mparkhe commented on mlflow/tracking/client.py:
```diff
@@ -208,7 +208,7 @@ def set_terminated(self, run_id, status=None, end_time=None):
         :param end_time: If not provided, defaults to the current time."""
         end_time = end_time if end_time else int(time.time() * 1000)
-        status = status if status else "FINISHED"
+        status = status if status else RunStatus.to_string(RunStatus.FINISHED)
```
flipping existing usage of free flowing strings to pre-defined constants. ^^this^^ is one example. More further down..

sueann commented on mlflow/tracking/fluent.py:
```diff
@@ -150,7 +150,7 @@ def start_run(run_uuid=None, experiment_id=None, source_name=None, source_versio
 
 
-def end_run(status="FINISHED"):
+def end_run(status=RunStatus.to_string(RunStatus.FINISHED)):
```
n00b questions for you: is the goal to have functions take in the enum and not the string? seems true for create_run above but not for this one (end_run). or these functions are meant for different contexts so we want the enum for one and the string for the other ?

sueann replied:
but we do want this `end_run` function to take in a string because it's exposed to the end user, right? (so this PR is good to go?) and eventually we want all the _internal_ methods to take in the enum.

mparkhe replied:
Yeah. I agree that user facing functions to take in string-y fields and we can check for valid values.

This change only replaces the default value of status from constant `"FINISHED"` to `RunStatus.to_string(RunStatus.FINISHED)` <-- still a string argument.

sueann commented:
LGTM

> On Jan 3, 2019, at 4:53 PM, Mani Parkhe <notifications@github.com> wrote:
> 
> @mparkhe commented on this pull request.
> 
> In mlflow/tracking/fluent.py:
> 
> > @@ -150,7 +150,7 @@ def start_run(run_uuid=None, experiment_id=None, source_name=None, source_versio
>      return _active_run_stack[-1]
>  
>  
> -def end_run(status="FINISHED"):
> +def end_run(status=RunStatus.to_string(RunStatus.FINISHED)):
> Yeah. I agree that user facing functions to take in string-y fields and we can check for valid values.
> 
> This change only replaces the default value of status from constant "FINISHED" to RunStatus.to_string(RunStatus.FINISHED) <-- still a string argument.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or mute the thread.

Commit:
Some more missing parity features with FileStore

- runs.source_version column increased size to 50
- runs.name: column removed uniqueness constraint, defaults to "" (in entities.run_info) if not provided
- missing default system tags to match FileStore

Commit:
Test for SqlTags + 'mlflow.parentRunId' tag
